tests/test_read_configs_include.py

```diff
@@ -83,17 +83,21 @@ class TestReadConfigs:
     ]
 
     def run_func(
-            self, include_file: str, local_env: Union[str, dict], command_suffix: str = ""
+            self, include_file: str, local_env: Union[str, dict], command_suffix: str = "", exit_on_error: bool = True
     ) -> subprocess.CompletedProcess:
         try:
             with tempfile.NamedTemporaryFile(delete=False, mode="w") as f:
                 set_local_env(f, local_env)
 
-            env = {"BIRDHOUSE_LOCAL_ENV": f.name}
+            env = {**os.environ, "BIRDHOUSE_LOCAL_ENV": f.name}
 
+            command_sequence = [f". {include_file}", "read_configs"]
             if command_suffix:
-                command_suffix = f"&& echo '{ENV_SPLIT_STR}' && {command_suffix}"
-            command = f". {include_file} && read_configs {command_suffix}"
+                command_sequence.extend([f"echo '{ENV_SPLIT_STR}'", f"{command_suffix}"])
+            if exit_on_error:
+                command_sequence.insert(1, "set -e")
+                command_sequence.insert(3, "set +e")
+            command = " ; ".join(command_sequence)
```

Comment:
All tests cases omit `exit_on_error`.
So basically, the `set` commands are always added since the default is `True`.

Does that mean that `set -e` should always be added to make sure that however we edit the scripts, this special case is always considered? 

Why not have a test that validates that `set -e` is always applied in the files instead? This way, we make sure the results from actually running the scripts and evaluating them in the tests do the same thing.

Reply:
> So basically, the set commands are always added since the default is True.

Yes, that way we can use existing tests to ensure that there isn't some line we're missing.

> Does that mean that set -e should always be added to make sure that however we edit the scripts

No I don't think that is necessary. `set -e` is sort of like running on strict mode, so if you run a script without it you should always get fewer errors, not more or different errors.

I like giving the scripts the flexibility of getting to choose whether to run with `set -e` or not. All this does is ensure that if they *do* run with `set -e` there won't be any unexpected errors.

Reply:
> `set -e` is sort of like running on strict mode, so if you run a script without it you should always get fewer errors

Exactly. Under which condition we wouldn't want those reported as early as possible?

The way I see it, if this test is merged, all scripts evaluated in the tests will be strictly validated against `set -e` being enabled. Therefore, the scripts might as well include it directly so that running them on the deployments has the same behavior as evaluated here. If not applied, we could get situations where an error that *would* have been caught, will fail silently, and we'll be confused as to why the tests cannot replicate the issue.

Reply:
Ok, should we just throw a `set -e` into the read-configs.include.sh file itself then?

Reply:
Well, I am not against `set -e` everywhere but that means we should test **all** the scripts whether we need to add additional `|| true` in there !  Quite an effort.  If we really want this, I'd suggest in a separate PR.

The `set -e` here in the test only test the `read_configs` fonction works correctly with `set -e`.  It is not testing weather all the other scripts will function properly, under all unexpected circumstances, after consuming the `read_configs`.

Thinking further, enforcing `set -e` in `read-configs.include.sh` also force all external scripts in external repos that source `read-configs.include.sh` to also test for all unexpected conditions (ex: missing matching `__DEFAULT__` vars) and add  `|| true`  where needed.  That's could potentially be quite de-stabilizing.  Not sure it is worth the effort.

Reply:
I think external scripts sourcing `read-configs.include.sh` would somewhat expect that our own definitions are valid and don't cause any unhandled case. However, I'm fine with doing it as a separate PR to evaluate the feature on its own.